python/acron/client/auth.py

```diff
@@ -60,26 +66,16 @@ def ask_for_secret():
     username = get_user_from_principal()
     secret = None
     while secret is None:
-        print("""Login for %s:
-
-1. Authenticator App
-2. Yubikey
-
-Option (1-2): """ % username, end="")
+        print("Your 2nd factor (%s):" % username)
         try:
             chars = sys.stdin.readlines(1)[0].rstrip()
-            if len(chars) == 6:
-                secret = otp_secret(chars)
-            if len(chars) == 44:
-                secret = yubi_secret(chars)
-            if chars == '1':
-                print('\nOTP: ', end="")
-                secret = otp_secret(read_secret(6))
-            if chars == '2':
-                print('\nYubikey: ', end="")
-                secret = yubi_secret(read_secret(44))
-            if secret is not None:
-                return secret
+            if len(chars) == 6 or len(chars) == 44:
+                if len(chars) == 6:
+                    secret = otp_secret(chars)
+                if len(chars) == 44:
+                    secret = yubi_secret(chars)
+                if secret is not None:
+                    return secret
             print('\nNo or invalid secret entered.')
```

Comment:
```suggestion
            if len(chars) != 6 and len(chars) != 44:
                print('\nNo or invalid secret entered.')

            if len(chars) == 6:
                secret = otp_secret(chars)
            if len(chars) == 44:
                secret = yubi_secret(chars)
            if secret is not None:
                return secret
```

@schwicke I'd suggest using a guard clause to reduce the indentation of the nested if-statements and, thus, reduce the complexity of the snippet.